### PR TITLE
v1.14 Backports 2023-11-07

### PR DESCRIPTION
**Pull request by jibi**
* [x] #28840 (@aanm)
 * [ ] #28787 (@universam1)
 * [x] #28808 (@brb)
 * [x] #28884 (@julianwiedmann)
 * [ ] #28923 (@jrajahalme)
 * [x] #28704 (@nathanjsweet)
 * [x] #28898 (@pchaigno)
 * [x] #28941 (@pchaigno)
 * [x] #28886 (@mhofstetter)
 * [x] #28927 (@sayboras)
 * [x] #28745 (@giorio94)
 * [x] #28966 (@tklauser)
 * [x] #29006 (@tgraf) 
 * [x] #28642 (@pchaigno)
 * [x] #28795 (@pchaigno)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 28840 28787 28808 28884 28923 28704 28898 28941 28886 28927 28745 28966 29006 28642 28795
```

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
My change looks good, thanks!

**Review by mhofstetter — APPROVED** [review]

**Comment by jibi** [comment]
/test-backport-1.14

File: daemon/cmd/state.go [hunk 1]
```diff
@@ -455,30 +457,49 @@ func (d *Daemon) initRestore(restoredEndpoints *endpointRestoreState, endpointsR
...
+				// running. Once this controller succeeds, because it has no
+				// RunInterval specified, it will not run again unless updated
+				// elsewhere. This means that if, for instance, a user manually
+				// adds a service via the CLI into the BPF maps, it will
+				// not be cleaned up by the daemon until it restarts.
+				syncServices := func(localOnly bool) {
+					d.controllers.UpdateController(
+						"sync-lb-maps-with-k8s-services",
+						controller.ControllerParams{
+							Group: syncLBMapsControllerGroup,
```
Comment (jibi) [inline daemon/cmd/state.go]: @giorio94 this is breaking the build as it's not defined in v1.14, the original controller down below was just:
```go
			controller.NewManager().UpdateController("sync-lb-maps-with-k8s-services",
					controller.ControllerParams{
						DoFunc: func(ctx context.Context) error {
							return d.svc.SyncWithK8sFinished(d.k8sWatcher.K8sSvcCache.EnsureService)
						},
						Context: d.ctx,
					},
				)
```

can I just skip the group/do I need additional commits?

Comment (giorio94) [reply daemon/cmd/state.go]: Yes, it is fine to drop the Group field (it only serves observability purposes, and it was not present in v1.14)

**Comment by jibi** [comment]
/test-backport-1.14

**Review by giorio94 — APPROVED** [review]
My PR looks good. Thanks!

**Review by aanm — APPROVED** [review]
LGTM for Thomas' PR

**Comment by jibi** [comment]
/test-backport-1.14

**Review by julianwiedmann — APPROVED** [review]

**Comment by pchaigno** [comment]
/test-backport-1.14

**Review by sayboras — APPROVED** [review]
Looks good for my commit ✅

**Review by squeed — APPROVED** [review]

**Review by brb — APPROVED** [review]
Thanks!

**Review by pchaigno — APPROVED** [review]
My two PRs you backported look good to me. Thanks!

**Review by nathanjsweet — APPROVED** [review]

**Pull request merged**